src/WebJobs.Extensions.DurableTask/Scale/DurableTaskTriggersScaleProvider.cs

```diff
@@ -61,6 +65,26 @@ public DurableTaskTriggersScaleProvider(
                 this.options.HubName);
         }
 
+        private static string? GetConnectionName(IDurabilityProviderFactory durabilityProviderFactory, DurableTaskOptions options)
+        {
+            if (durabilityProviderFactory is AzureStorageDurabilityProviderFactory azureStorageDurabilityProviderFactory)
+            {
+                // First, look for the connection name in the options initialized from the trigger metadata
+                var azureStorageOptions = new AzureStorageOptions();
+                if (options != null && options.StorageProvider != null)
+                {
+                    JsonConvert.PopulateObject(JsonConvert.SerializeObject(options.StorageProvider), azureStorageOptions);
+                }
+
+                // If the connection name is not found in the options, use the default connection name from the factory
+                return azureStorageOptions.ConnectionName ?? azureStorageDurabilityProviderFactory.DefaultConnectionName;
```

Comment:
I'm not understanding the behavior change. Doesn't the `AzureStorageDurabilityProviderFactory` constructor already do this exact logic and provide the correct storage account connection name in the `DefaultConnectionName` property?

Reply:
@cgillum The `AzureStorageDurabilityProviderFactory` does have the same logic, but it receives a different input, so it produces a different result. When this constructor is invoked _in the Scale Controller context_, the `options` parameter contains an empty `StorageProvider` collection. I don't yet understand why, and, more importantly, whether this is intentional or not. I suspect this has something to do with how DI instantiates these objects, and this may require a fix in the Scale Controller.

However, when the `DurableTaskTriggersScaleProvider` is invoked, the `triggerMetadata` parameter does contain correct `StorageProvider` collection, and this is why my fix works.